apps/els_lsp/src/els_code_actions.erl

```diff
@@ -0,0 +1,120 @@
+                        , els_protocol:range(#{from => Pos, to => Pos})) ]
+  end.
+
+-spec ignore_variable(uri(), range(), [binary()]) -> [map()].
+ignore_variable(Uri, Range, [UnusedVariable]) ->
+  {ok, Document} = els_utils:lookup_document(Uri),
+  POIs = els_poi:sort(els_dt_document:pois(Document, [variable])),
+  case ensure_range(els_range:to_poi_range(Range), UnusedVariable, POIs) of
+    {ok, VarRange} ->
+      [ make_edit_action( Uri
+                        , <<"Add '_' to '", UnusedVariable/binary, "'">>
+                        , ?CODE_ACTION_KIND_QUICKFIX
+                        , <<"_", UnusedVariable/binary>>
+                        , els_protocol:range(VarRange)) ];
+    error ->
+      []
+  end.
+
+-spec suggest_variable(uri(), range(), [binary()]) -> [map()].
+suggest_variable(Uri, Range, [Var]) ->
+  %% Supply a quickfix to replace an unbound variable with the most similar
+  %% variable name in scope.
+  {ok, Document} = els_utils:lookup_document(Uri),
+  POIs = els_poi:sort(els_dt_document:pois(Document, [variable])),
+  case ensure_range(els_range:to_poi_range(Range), Var, POIs) of
+    {ok, VarRange} ->
+      ScopeRange = els_scope:variable_scope_range(VarRange, Document),
+      VarsInScope = [atom_to_binary(Id, utf8) ||
+                      #{range := R, id := Id} <- POIs,
+                      els_range:in(R, ScopeRange),
+                      els_range:compare(R, VarRange)],
+      case [{els_utils:levenshtein_distance(V, Var), V} ||
+             V <- VarsInScope,
+             V =/= Var,
+             binary:at(Var, 0) =:= binary:at(V, 0)]
+      of
+        [] ->
+          [];
+        VariableDistances ->
+          {_, SimilarVariable} = lists:min(VariableDistances),
+          [ make_edit_action( Uri
+                            , <<"Did you mean '", SimilarVariable/binary, "'?">>
+                            , ?CODE_ACTION_KIND_QUICKFIX
+                            , SimilarVariable
+                            , els_protocol:range(VarRange)) ]
+      end;
+    error ->
+      []
+  end.
+
+-spec fix_module_name(uri(), range(), [binary()]) -> [map()].
+fix_module_name(Uri, Range0, [ModName, FileName]) ->
+  {ok, Document} = els_utils:lookup_document(Uri),
+  POIs = els_poi:sort(els_dt_document:pois(Document, [module])),
+  case ensure_range(els_range:to_poi_range(Range0), ModName, POIs) of
+    {ok, Range} ->
+      [ make_edit_action( Uri
+                        , <<"Change to -module(", FileName/binary, ").">>
+                        , ?CODE_ACTION_KIND_QUICKFIX
+                        , FileName
+                        , els_protocol:range(Range)) ];
+    error ->
+      []
+  end.
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+
+-spec ensure_range(poi_range(), binary(), [poi()]) -> {ok, poi_range()} | error.
```

Comment:
As per our offline discussion, this function is worth a comment, since it's used for two main reasons:

* No column numbers before OTP 24
* Handling a situation where diagnostics are not up-to-date with a modified buffer